modtools/create-item.lua

```diff
@@ -326,16 +336,27 @@ function hackWish(accessors, opts)
         until count
     end
     if not mattype or not itemtype then return end
-    if df.item_type.attrs[itemtype].is_stackable then
+    if not typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] and df.item_type.attrs[itemtype].is_stackable then
         return createItem({mattype, matindex}, {itemtype, itemsubtype}, quality, unit, description, count)
     end
+    if typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] and df.item_type.attrs[itemtype].is_stackable then
+        return createItem({matindex, casteId}, {itemtype, itemsubtype}, quality, unit, description, count)
+    end
```

Comment:
There is too much duplication here. How about this?
```lua
    if df.item_type.attrs[itemtype].is_stackable then
        local mat = typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] and {matindex, casteId} or {mattype, matindex}
        return createItem(mat, {itemtype, itemsubtype}, quality, unit, description, count)
    end
```

Reply:
Wait, did you test this?

Reply:
not tested, just written as a suggestion for code improvement. If you have two chunks of code that are semantically and structurally identical except for a small difference, it is usually better to factor that difference out and keep the structure simpler.